app/generate_client_secrets.py

```diff
@@ -0,0 +1,24 @@
+'''
+Generate a client_secrets.json based on environment variables.
+This is a workaround for the Google oauth library not being able
+to read in credentials *except* from JSON source... yay!
+'''
+
+import os
+import json
+
+client_secrets = {
+  'web': {
+    'client_id': os.getenv('client_id', 'no id set'),
+    'client_secret': os.getenv('client_secret', 'no secret set'),
+    'redirect_uris': os.getenv('redirect_uris', 'http://localhost:0').split(','),
+    'auth_uri': os.getenv('auth_uri', 'http://localhost:0'),
+    'token_uri': os.getenv('token_uri', 'http://localhost:0')
+  }
+}
+
+data = json.dumps(client_secrets)
+
+text_file = open("client_secrets.json", "w")
```

Comment:
not strictly heroku, but yes: we don't want to check in a client_secrets.json, and the auth library does not read in values from env, so we need a little script that can generate the file the library wants, based on variables in the active environment